wsgi.py

```diff
@@ -6,255 +6,348 @@
-                response=json.dumps({"msg": "unable to create openshift user identity mapping (" + user_name + ")"}),
+                response=json.dumps(
+                    {
+                        "msg": "unable to create openshift user identity mapping ("
+                        + user_name
+                        + ")"
+                    }
+                ),
                 status=400,
-                mimetype='application/json'
-                )
+                mimetype="application/json",
+            )
+    else:
+        user_exists = user_exists | 0x04
 
-    if user_exists and identity_exists and mapping_exists:
+    if user_exists == 7:
         return Response(
             response=json.dumps({"msg": "user currently exists (" + user_name + ")"}),
             status=200,
-            mimetype='application/json'
-            )
+            mimetype="application/json",
+        )
     return Response(
         response=json.dumps({"msg": "user created (" + user_name + ")"}),
         status=200,
-        mimetype='application/json'
-        )
-
-
-@application.route("/users/<user_name>", methods=['DELETE'])
-def delete_moc_user(user_name, full_name=None, id_provider="sso_auth", id_user=None):
-    token, openshift_url = get_token_and_url()
-
-    user_exists = exists_openshift_user(token, openshift_url, user_name)
-    if user_exists:
-        r = delete_openshift_user(token, openshift_url, user_name, full_name)
-        if r.status_code not in [200, 201]:
+        mimetype="application/json",
+    )
+
+
+@application.route("/users/<user_name>", methods=["DELETE"])
+@auth.login_required
+def delete_moc_user(user_name):
+    shift = get_openshift()
+    r = None
+    user_does_not_exist = 0
+    # use case if User exists then delete
+    if shift.user_exists(user_name):
+        r = shift.delete_user(user_name)
+        if r.status_code != 200 and r.status_code != 201:
             return Response(
-                response=json.dumps({"msg": "unable to delete User (" + user_name + ") 1"}),
+                response=json.dumps(
+                    {"msg": "unable to delete User (" + user_name + ") 1"}
+                ),
                 status=400,
-                mimetype='application/json'
-                )
-
-    if id_user is None:
-        id_user = user_name
-
-    identity_exists = exists_openshift_identity(token, openshift_url, id_provider, id_user)
-    if identity_exists:
-        r = delete_openshift_identity(token, openshift_url, id_provider, id_user)
-
-        if r.status_code not in [200, 201]:
+                mimetype="application/json",
+            )
+    else:
+        user_does_not_exist = 0x01
+
+    # This is the specific business case for the MOC sort of
+    # TODO: generalize this in the next version.
+    #    1) get the list of identities associated with the user
+    #    2) delete each one (as well as doing this part first)
+    id_user = user_name
+    id_provider = "sso_auth"
```

Comment:
Elsewhere you default `id_provider` to `moc-sso`. We should probably be consistent.

Reply:
k-openshift defaults to sso_auth, you choose moc-auth.  Will be changed.